packages/dashboard/src/components/DashboardGridLayout/AddCardPanel/AddCardPanelBase.tsx

```diff
@@ -0,0 +1,204 @@
+import * as React from 'react';
+
+import { Panel, PanelType } from 'office-ui-fabric-react/lib/Panel';
+import { classNamesFunction, BaseComponent, getRTL } from 'office-ui-fabric-react/lib/Utilities';
+
+import { AddCard } from './AddCard/AddCard';
+import { getStyles } from './AddCardPanel.styles';
+import { IAddCardPanelProps, IAddCardPanelState, IAddCardPanelStyles, IAddCardPanelStyleProps } from './AddCardPanel.types';
+import { IDGLCard, CardSize, DraggingAnimationType } from '../../../index';
+import { ITheme } from 'office-ui-fabric-react/lib/Styling';
+
+export class AddCardPanelBase extends BaseComponent<IAddCardPanelProps, IAddCardPanelState> {
+  constructor(props: IAddCardPanelProps) {
+    super(props);
+    this.state = {
+      flyoutStyle: {}
+    };
+  }
+
+  public render(): JSX.Element {
+    const { header, isOpen, cards, closeButtonAriaLabel, theme } = this.props;
+    return (
+      <>
+        <Panel
+          closeButtonAriaLabel={closeButtonAriaLabel}
+          onRenderBody={this._renderAddCardItems.bind(this, cards, header, isOpen, theme)}
+          type={PanelType.custom}
+          customWidth={'480px'}
+          isOpen={isOpen}
+          isBlocking={false}
+          onDismiss={this._onDismiss}
+          // tslint:disable-next-line:jsx-ban-props
+          style={this.state.flyoutStyle}
+        />
+      </>
+    );
+  }
+
+  private _onDismiss = () => {
+    if (this.props.onDismiss) {
+      this.props.onDismiss();
+    }
+  };
+
+  private _getEmptyCard = (): string => {
+    const emptyCardArray = [
+      './src/images/AddCardPanelEmptyState/horizontalBarChar.svg',
+      './src/images/AddCardPanelEmptyState/lineGraph.svg',
+      './src/images/AddCardPanelEmptyState/verticalBarChart.svg'
+    ];
+    const randomCard = emptyCardArray[Math.floor(Math.random() * emptyCardArray.length)];
+    return randomCard;
+  };
+
+  private _renderAddCardItems = (addCardItems: IDGLCard[], header: string, isOpen: boolean, theme: ITheme): JSX.Element => {
```

Comment:
Just a tip, any time you see a _renderFoo function, it should be a hint that maybe you should break out a subcomponent; smaller things are easier to maintain.